File: src/main/java/com/qtech/bubbles/common/Listener.java
package com.qtech.bubbles.common;

/**
 * Event listener.
 * <p>
 * Use {@link #bindEvents()} for binding events to method(s).
 * And use {@link #unbindEvents()} for unbind events from method(s).
 * <p>
 * Also in the (un)bind methods assign a variable to true if the events were binded, false otherwise.
 * And last, use {@link #areEventsBound()} for the boolean assigned in the bind/unbind events methods.
 */
public abstract class Listener {
    protected boolean eventsActive;

    protected abstract void bindEvents();

    protected abstract void unbindEvents();

    protected abstract boolean areEventsBound();
}

File: src/main/kotlin/com/qtech/bubbles/common/Listener.kt
package com.qtech.bubbles.common

/**
 * Event listener.
 *
 *
 * Use [.bindEvents] for binding events to method(s).
 * And use [.unbindEvents] for unbind events from method(s).
 *
 *
 * Also in the (un)bind methods assign a variable to true if the events were bound, false otherwise.
 * And last, use [.areEventsBound] for the boolean assigned in the bind/unbind events methods.
 */
abstract class Listener {
    protected var eventsActive = false
    protected abstract fun bindEvents()
    protected abstract fun unbindEvents()
    protected abstract fun areEventsBound(): Boolean
}